### ISSUE TITLE
How to set different Aex,DMI and anistropy constant in different region?

### ISSUE THREAD

**Issue by Donghi007** [MagneticSimulation/MicroMagnetic.jl#12]
Dear developers,

Thank you for developing this excellent package. It's very nice to use it into my work. However, I encounter some issues regarding the simulation regions in `MicroMagnetic.jl`

My question is that how to define region with different simulation parameters? Let's say there has a simulation box which has nx=ny=100, nz=4 nodes along x,y,z (totally 4 layers). Now I hope to set  A=A1, DMI=D1, K=K1 for the layer nz from 1-2  and  A=A2, DMI=D2, K=K2 for the layer nz from 3-4.  But it's difficult for me to do that.  I hope to know whether there is a method can do this.

I am truly appreciated for your help :)

Best,

**Comment by ww1g11** [comment]
Hi @Donghi007 

It's straightforward to set spatial parameters in MicroMagnetic.jl. For instance, the exchange constant for different layers can be specified by defining a custom function. Here’s an example from the [documentation](https://magneticsimulation.github.io/MicroMagnetic.jl//dev/api/#MicroMagnetic.add_exch-Tuple%7BMicroMagnetic.MicroSim,%20Union%7BFunction,%20Number,%20Tuple,%20Array%7D%7D):
```
function spatial_A(i, j, k, dx, dy, dz)
    if k <= 2
        return 1e-11  # Exchange constant A1 for layer 1 and layer 2
    else
        return 2e-11  # Exchange constant A2 for other regions
    end
end
add_exch(sim, spatial_A)
```
Other parameters can be set in a similar manner.

Best wishes,

**Comment by Donghi007** [comment]
Dear @ww1g11 ,

Very thanks for you help!

At first, I was very confused that why functions such like 'spatial_()' can be called without defining any 'i,j,k' in the online 'MicroMagnetic.jl' tutorial. And function 'spatial_()' only return one float/double type value, so why function 'add_exch()' can set A at different site (i,j,k). 

Enlightened by your previous helpful example code, now I understand that maybe in the inside of 'add_exch()', there has defined the method how to call 'spatial_()' and 'i,j,k'.

Thanks again for you help:)

Best,

**Comment by ww1g11** [comment]
Yes, in the code we call the functions provided by the user, please see lines 57-63 in the file hepler.jl at https://github.com/MagneticSimulation/MicroMagnetic.jl/blob/master/src/helper.jl
